Reference_Manual.md

```diff
@@ -3287,6 +3295,262 @@ This typed_data allows you to retrieve the data type of the underlying binary to
+         .expected_result_type(sql_expected_result_type::rows); // Expect a table which contains rows
+
+// To fetch use non-parameterized overloads
+std::cout << "cursor_buffer_size : "   << statement.cursor_buffer_size()        << std::endl
+          << "timeout : "              << statement.timeout().count()           << std::endl
+          << "expected_result_type : " << int(statement.expected_result_type()) << std::endl;
+
+hz.get_sql().execute(statement); // OK, execute this statement
+```
+
+### 7.11.5 Read Row Metadata
+Row metadatas can be acquired with `sql_result::row_metadata()` and `sql_page::sql_row::row_metadata()` method.
+
+``` C++
+using namespace hazelcast::client::sql;
+
+auto hz = hazelcast::new_client().get();
+
+auto result = hz.get_sql().execute("SELECT name, surname, age FROM employees").get();
+
+auto metadata = result->row_metadata();
+
+// Print column count
+std::cout << "There are " << metadata.column_count() << " columns" << std::endl;
+
+// Print columns
+for (const sql_column_metadata& col : metadata.columns())
+{
+    std::cout << "name : "      << col.name
+              << " type : "     << int(col.type)
+              << " nullable : " << col.nullable
+              << std::endl;
+}
+
+// Get column by index
+auto age_col_metadata = metadata.column(2);
+
+std::cout << "name : "      << age_col_metadata.name
+          << " type : "     << int(age_col_metadata.type)
+          << " nullable : " << age_col_metadata.nullable
+          << std::endl;
+```
+**Note :** There is no metadata for non-`SELECT` queries.
+
+``` C++
+auto result = sql.execute("DELETE FROM employees WHERE age < 18").get();
+
+result->row_metadata(); // Throws an `exception::illegal_state`
+                        // This is not a SELECT query
+```
+
+### 7.11.6 Read Table Cell Value
+`boost::optional<T> sql_page::sql_row::get_object<T>` is used for reading cell. `T` specifies the return type and it should correspond with the column type. Otherwise it will throw `boost::bad_any_cast`. There are two overloads one can be used to read value by column idx and other by column name.
+
+``` C++
+for (auto itr = result->iterator(); itr.has_next();)
+{
+    auto page = itr.next().get();
+
+    for(const sql_page::sql_row& row : page->rows())
+    {
+        std::cout << row.get_object<int>(0) // Read first column value as int
+                  << ", "
+                  << row.get_object<int>("age") // Read 'age' column value
+                  << ", "
+                  << row.get_object<std::string>("surname") // Read 'surname' column value
+                  << std::endl;
+    }
+}
+```
+
+### 7.11.7 Error Handling
+There are some exceptions which expresses an error occurred while performing an operation.
+So those exceptions should be taken into consideration for production ready code.
+
+`page_iterator::next()` can throw three types of exceptions.
```

Comment:
It can actually throw a lot more than these, right? We are not wrapping all exceptions to sql_exception, right?

Reply:
Fixed.

Reply:
unfixed